usr/share/rear/prep/Linux-i386/370_detect_secure_boot.sh

```diff
@@ -0,0 +1,34 @@
+test "$SECURE_BOOT_BOOTLOADER" && return 0
+
+# Note:
+# this is more of a hack than a really good solution.
+# The good solution would check the EFI variables for the bootloader
+# that is used to boot the system. But this is not implemented yet for
+# secure boot.
+#
+# The code in usr/share/rear/rescue/default/850_save_sysfs_uefi_vars.sh
+# could be used as a starting point for this, however currently
+# it tries to read the actual boot loader# used only as a last resort
+# if well-known files are not found.
+#
+if type -p mokutil ; then
+    PROGS+=( mokutil )
+    local secureboot_status
+    if secureboot_status=$(mokutil --sb-state 2>&1) ; then
```

Comment:
I considered doing that as well but I decided against it. The problem is, distributions often use the shim even on non-secure-boot enabled installs, presumably to reduce variations in possible configurations and to allow turning on Secure Boot later. But, if shim is used and Secure Boot is not currently enabled, the current code (and also your suggested change) will produce a recovered system that does not use shim but boots Grub directly. This is against the goal of ReaR recovery to reproduce the original system as faithfully as possible, and creates a hidden problem when the user decides to turn on Secure Boot in the recovered system later and the system fails to boot.

Reply:
Yes, I thought about that. I'm not sure if we need ReaR as a "migrate to secure boot tool".

I agree with you that recovering a system as it was before is the goal. I'll test a full recovery to see if indeed my code creates a degraded recovery or not.

Reply:
I think you misunderstood. I don't mean that ReaR would be used as a "migrate to secure boot tool". I merely meant that if EFI Shim is in use, ReaR should preserve it, even if Secure Boot is not enabled.

Yes, try a full recovery of a system that uses Shim but has Secure Boot disabled. That said, current ReaR has the same problem (the recovered system does not use Shim anymore), so your approach does not make it worse.

Reply:
I see. Yes. In that case I'd still suggest to merge this here sooner than later because it improves ReaR for the user (no need to manually configure secure boot shim).

And yes, we should fix secure boot properly. Given the tremendous complexity of our boot handling I'm more and more wondering if we shouldn't do something new from scratch with less scope and ask users to use older ReaR for systems that don't support the new method.

Reply:
@pcahyna I can confirm that also with this code a recovered system that was using the shim without enforcing secure boot gets recoverd without the shim.